core/src/OpenMPTarget/Kokkos_OpenMPTarget_Exec.hpp

```diff
@@ -1205,22 +1203,14 @@ KOKKOS_INLINE_FUNCTION void parallel_reduce(
     const Impl::ThreadVectorRangeBoundariesStruct<
         iType, Impl::OpenMPTargetExecTeamMember>& loop_boundaries,
     const Lambda& lambda, ValueType& result) {
-  result = ValueType();
-
-  ValueType* tmp_scratch =
-      (ValueType*)loop_boundaries.team.impl_reduce_scratch();
-#pragma omp barrier
-  tmp_scratch[1] = ValueType();
-#pragma omp barrier
-
-#pragma omp simd reduction(+ : tmp_scratch [1:2])
+  ValueType vector_reduce = ValueType();
+#pragma omp simd reduction(+ : vector_reduce)
```

Comment:
Consider renaming variable to scalar_reduce as this is for the scalar-type reduction.

Reply:
Yeah it is confusing I guess. The reason I named it as `vector_reduce` is because the reduction is happening over the ThreadVectorRange. I can rename it if you push me strongly over it :-) .